### PR TITLE
feat: add long stay baseline work

### PR DESCRIPTION
**Pull request by amadeuspzs**
Add write up of Length of Stay baseline models

### CODE REVIEW

File: documentation/long-stay-baseline.mdx [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+3. Simple baseline models implemented using commonly available packages including [scikit-learn 1.1.1](https://scikit-learn.org/), [CatBoost 1.0.6](https://catboost.ai) and [XGBoost 1.3.3](https://xgboost.readthedocs.io/en/stable/)
+4. Analysis of model performance by demographic
+5. A comparison of regression-based and classification-based risk stratification models
+6. A set of extensions for future work
+
+## 3. Data ingest and processing<a name='dataingestandprocessing'></a>
+
+GHNHSFT performed a SQL export from their EPR system containing ~770,000 records across 99 columns, with significant sparsity across several columns and a section of rows, as visualised by light coloured blocks (null values) in the below image:
+
+![Image of data sparsity](images/long-stay-baseline/sparsity.jpeg)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:62]: y-axis: the row information doesn't seems like too important. May be can change it to something like `case_id`/`case_index`? Also can reduce the number of y-ticks location (e.g., just show, 1, 100000, 200000, etc.). Now it looks a bit too many numbers listed on the y-axis.

File: documentation/long-stay-baseline.mdx [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+The population for this study was defined as non-elective, major cases as recorded in the `IS_MAJOR` and `elective_or_non_elective` fields.
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:64]: Are `IS_MAJOR` and `elective_or_non_elective` two of the columns in the data that shown here in the plot? May be they are in different name?

File: documentation/long-stay-baseline.mdx [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+
+Filtering the dataset to this definition resulted in a reduction of ~770,000 rows to ~170,000 rows (78% reduction):
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:66]: But the plot below `sparsity-major.jpeg` shows >645011 rows. The one plot after (`sparsity-clean.jpeg`) shows the correct number of rows (~170,000). From reading the name of the image files here I can know that the rows in `sparsity-major.jpeg` is for the major case only, but it's not too clear without knowing the image name.

File: documentation/long-stay-baseline.mdx [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+
+![Image of data sparsity - major cases](images/long-stay-baseline/sparsity-major.jpeg)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:68]: Same comment as the `sparsity.jpeg`, about the y-axis ticks.

File: documentation/long-stay-baseline.mdx [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+3. Removing fields not available at admission
+4. Removing empty and redundant (e.g. `LENGTH_OF_STAY_IN_MINUTES` duplicates `LENGTH_OF_STAY`) columns
+5. Removing duplicate rows
+4. Removing local identifiers
+5. Imputing `stroke_ward_stay` as `N` if not specified
+6. Binary encoding `stroke_ward_stay`, `IS_care_home_on_admission`, `IS_care_home_on_discharge` and `IS_illness_not_injury`
+
+This resulted in ~170,000 rows across ~50 columns as visualised in the below image:
+
+![Image of data sparsity of "clean" data](images/long-stay-baseline/sparsity-clean.jpeg)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:83]: Same comment as the `sparsity.jpeg`, about the y-axis ticks.

File: documentation/long-stay-baseline.mdx (hunk 1 above)
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:62]: Can also add some simple figure captions. Just like the text already in the `alt text` input (e.g., `[Image of data sparsity]` or `[Image of data sparsity - major cases]` for the later plot). The `alt text` input is not showing in the browser view.

File: documentation/long-stay-baseline.mdx [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+
+The resulting data dictionary is available [here](https://github.com/nhsx/skunkworks-long-stayer-risk-stratification-baseline/blob/main/docs/data-dictionary.csv).
+
+Additionally, Length of Stay was capped to 30 days, due to a long tail of long stayers over ~15 days and the definition of long stayer being over 21 days. The effect of capping can be visualised by comparing box plots of the distribution of length of stay on the raw data (left image, y scale up to 300 days) and the capped data (right image, y scale up to 30 days):
+
+<p align="center">![Boxplot of length of stay](images/long-stay-baseline/los-boxplot.png)</p>
+
+The resulting distribution of length of stays shows a ~bimodal distribution - the majority of stays are short (&lt;5 days), which a long tail and population of long stayers:
+
+<p align="center">![Density plot of length of stay](images/long-stay-baseline/los-density.jpeg)</p>
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:93]: The peak for the long stayer (>30 days):
- Worth mentioning the bimodal behaviour is caused by grouping all >30 days examples to 1 bin.
- x-axis ticks: should show > 30 days for that bin, instead of just 30 days.

File: documentation/long-stay-baseline.mdx [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+
+In order to select appropriate modelling approaches, some basic statistical analysis was conducted to understand normality and inter-correlation of the selected features.
+
+#### Correlation analysis
+
+Correlation analysis confirmed presence of significant **collinearity** between different features.
+
+Top 20 one-hot encoded features correlated with `LENGTH_OF_STAY`, ranked by absolute correlation, were:
+
+Feature|Correlation (Pearson correlation coefficient)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:152]: Is it possible to show in a histogram plot and/or list them in descending order w.r.t. the `Correlation (Pearson correlation coefficient)`?

File: documentation/long-stay-baseline.mdx [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+6. Tune the best performing model using cross-validation on the training and validation set
+7. Report the final performance of the model using the test set
+
+![Summary of machine learning approach](images/long-stay-baseline/ml-approach.png)
+
+Training, validation and test splits were representative of the population and did not introduce selection bias:
+
+**Length of stay**
+
+![Distribution of length of stay by data split](images/long-stay-baseline/split-los.jpeg)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:209]: x-axis ticks: should show > 30 days for that bin, instead of just 30 days. (Same as the `los-density.jpeg`)

File: documentation/long-stay-baseline.mdx [hunk 9]
```diff
@@ -0,0 +1,528 @@
...
+
+**Age**
+
+![Distribution of age by data split](images/long-stay-baseline/split-age.jpeg)
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:213]: - Histogram bin width should be the same for all 3 dataset. Different bin width can cause the distribution looks different.
- x- and y-label are missing.

File: documentation/long-stay-baseline.mdx [hunk 10]
```diff
@@ -0,0 +1,528 @@
...
+train: [0.53, 0.47]
+validate: [0.51, 0.49]
+test: [0.53, 0.47]
+```
+
+**Ethnicity**
+
+Proportions of each ethnicity for each split:
+
+```
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:229]: There're a lot of `0.0` in this proportions. Maybe can clean it a bit by just showing the first ~7 largest group?

File: documentation/long-stay-baseline.mdx [hunk 11]
```diff
@@ -0,0 +1,528 @@
...
+
+![Underlying counts for index of multiple deprivation - all data](images/long-stay-baseline/los-dist-imd.png)
+
+Index of Multiple Deprivation (IMD) deciles are skewed to the lower end, ie. there are more deprived patients present in this dataset than not.
+
+Now we can look at the distribution of length of stay for the above categories:
+
+![Underlying length of stay by ethnicity - all data](images/long-stay-baseline/los-mean-los-ethnicity.png)
+
+There is significant variation of length of stay for different ethnic groups, for example with White and black Carribean patients having a length of stay of 2.6 days on average, versus 6.0 days for Irish patients. However, as discussed previously, the count of these groups is 560 and 892 individuals respectively so further tests need to be conducted to understand if the distributions are truly different (e.g. a two-sided Kolmogorov test).
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:337]: ...so further ^statistical hypothesis^ tests need to be conducted...
(e.g. a two-sided ^Kolmogorov–Smirnov^ test)

File: documentation/long-stay-baseline.mdx [hunk 12]
```diff
@@ -0,0 +1,528 @@
...
+
+![Underlying length of stay by sex - all data](images/long-stay-baseline/los-mean-los-sex.png)
+
+Length of stay is almost identical across patient sex.
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:341]: ^Mean^ LoS...

File: documentation/long-stay-baseline.mdx [hunk 13]
```diff
@@ -0,0 +1,528 @@
...
+Decision Tree|[DecisionTreeRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeRegressor.html)|[DecisionTreeClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html)
+Random Forest|[RandomForestRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html)|[RandomForestClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html)
+XGBoost|[XGBRegressor](https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBRegressor)|[XGBClassifier](https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBClassifier)
+CatBoost|[CatBoostRegressor](https://catboost.ai/en/docs/concepts/python-reference_catboostregressor)|[CatBoostClassifier](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier)
+
+The training, validation and test regime was the same as for the regression models.
+
+**Class imbalance**
+
+In the training set, we observe the following class imbalance:
```
Comment (sansonpoon) [inline documentation/long-stay-baseline.mdx:396]: May be can also include the values for the validation set as well?

File: documentation/long-stay-baseline.mdx [hunk 14]
```diff
@@ -0,0 +1,528 @@
...
+
+<Tags title="" tags={['LoS','length of stay','baseline','risk model', 'regression', 'classification']} />
+
+> Baseline machine learning models using historical data from Gloucestershire Hospitals NHS Foundation Trust (GHNHSFT) to predict how long a patient will stay in hospital upon admission.
+
+Long Stayer risk stratification baseline models was selected as a project to run in tandem with the [Long Stayer Risk Stratification](long-stay) project, and started in March 2022.
+
+Baseline models provide a mechanism to generate baseline metrics to assess the performance of more complex models, and establish the effectiveness of simple approaches.
+
+<Alert title="Intended audience">This report has been written for analysts and data scientists at NHS Trusts/ALBs</Alert>
```
[the remark this answers is not in the document]
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx:15]: I did want to be explicit here as it's essentially a walkthrough of the data science pipeline

File: documentation/long-stay-baseline.mdx (hunk 3 above)
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx:66]: Good shout. I will reindex the dataframe just for comprehension

File: documentation/long-stay-baseline.mdx (hunk 2 above)
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx:64]: Not all columns are labelled due to the width of the dataframe. I can make this clearer

File: documentation/long-stay-baseline.mdx (hunk 1 above)
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx]: Great suggestion!

File: documentation/long-stay-baseline.mdx (hunk 9 above)
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx]: Good spot!

File: documentation/long-stay-baseline.mdx (hunk 13 above)
Comment (amadeuspzs) [reply documentation/long-stay-baseline.mdx:421]: Because the class weights are determined using the training data (so as not to leak information from the validation/test set) I think this is sufficient.

Added a sentence to clarify

**Review by sansonpoon — APPROVED** [review]
Multiple comments submitted and resolved.
